### DISCUSSION TITLE
Sync Actions 无法生成分析 SVG && 位置信息无法更新

### DISCUSSION THREAD

**Discussion by 85Ryan** [yihong0618/running_page#576] [Q&A] [answered]
昨天按 README 步骤完成了部署设置，遇到下面几个问题：

> REPO：https://github.com/85Ryan/gooorun 
> GH PAGES：https://85ryan.github.io/gooorun/
> 数据源：STRAVA

1、Sync Action 更新数据时，显示更新完成，但无法生成最新的 Total SVG 图片，返回提示如下：
```
  Run python run_page/gen_svg.py --from-db --title "Ryan's Running" --type github --athlete "Ryan" --special-distance 10 --special-distance2 20 --special-color yellow --special-color2 red --output assets/github.svg --use-localtime --min-distance 0.5
  All tracks: 28
  After filter tracks: [28](https://github.com/85Ryan/gooorun/actions/runs/7239220950/job/19720846244#step:21:29)
  Creating poster of type github with 26 tracks and storing it in file assets/github.svg...
  All tracks: 28
  After filter tracks: 28
  All tracks: 28
  After filter tracks: 28
  All tracks: 28
  After filter tracks: 28
  Creating poster of type github with 26 tracks and storing it in file assets/github_2023.svg...
  Cannot set locale to "zh_CN": unsupported locale setting
```

2、LocationSummary 无法获取位置统计信息
<img width="608" alt="截屏2023-12-18 08 36 36" src="https://github.com/yihong0618/running_page/assets/3353632/8d730537-491e-4098-b86d-69446e274224">

3、我自己替换修改了 assets 文件夹下的地图开始点、结束点图标：start.svg 和 end.svg，本地显示正常，部署到 Github Pages 后不能正常显示，还是显示之前的图标。

以上几个问题，一直没有排查到原因，能否帮忙检查一下，谢谢！

**Comment by yihong0618** [comment, marked as answer]
好的。
1. 你可以完全用 GitHub Actions 跑，本地跑获取地点需要梯子
2. 关于 Total SVG 应该也是你本地没有 locale 的问题

**Reply by 85Ryan** [reply]
好的，感谢，我试一下。

完全用 Actions 跑的话，是不是直接 `pnpm data:clean` 清空本地数据，然后 PUSH 就可以了？

**Reply by yihong0618** [reply]
不是，之际新项目或者 fork 然后直接走 GitHub Actions 那一步

**Reply by 85Ryan** [reply]
好的，我试一下，谢谢

**Reply by 85Ryan** [reply]
重新 Fork 了一个，完全 Actions 执行，LocationSummary 显示正常，👍  

但在生成 SVG 图片时，但最后还是提示：`Cannot set locale to "zh_CN": unsupported locale setting`，不知道是什么原因。

**Reply by yihong0618** [reply]
@85Ryan 不用中文试试

**Reply by 85Ryan** [reply]
我将生成 Github SVG 的代码：

```
python run_page/gen_svg.py --from-db --year $(date +"%Y")  --language zh_CN --title "$(date +"%Y") Running" --type github --athlete "${{ env.ATHLETE }}" --special-distance 10 --special-distance2 20 --special-color yellow --special-color2 red --output assets/github_$(date +"%Y").svg --use-localtime --min-distance 0.5
``` 
其中的参数 ` --language zh_CN` 去掉，就正常了 🤣

**Comment by 85Ryan** [comment]
尝试在之前项目上解决这个问题，发现删除项目目录下  `run_page/data.db` 与 `src/static/activities.json` 两个本地生成的数据文件，push 后用 actions 重新跑一遍，就能解决不能生成 `LocationSummary` 的问题，同时也能生成正确的 SVG 图。

另外，我发现运行 Actions 时，只能生成新的 `activities.json` 文件，而并不能生成（覆盖）旧的 `data.db` 文件，同时生成 SVG 又是依赖于 `data.db` 中的数据，所以导致 SVG 文件不能随着 `activities.json` 更新而同步更新。

（不过，我也不能确定是不是这个原因....😃）

**Reply by yihong0618** [reply]
rm imported.json
rm run_page/data.db
rm GPX_OUT/*
